account.go

```diff
@@ -0,0 +1,160 @@
+package caigo
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"net/http"
+)
+
+type Signer struct {
+	private *big.Int
+	Curve   StarkCurve
+	Gateway *StarknetGateway
+	PublicX *big.Int
+	PublicY *big.Int
+}
+
+type FeeEstimate struct {
+	Amount *big.Int `json:"amount"`
+	Unit   string   `json:"unit"`
+}
+
+/*
+	Instantiate a new StarkNet Signer which includes structures for calling the network and signing transactions:
+	- private signing key
+	- stark curve definition
+	- full StarknetGateway definition
+	- public key pair for signature verifications
+*/
+func (sc StarkCurve) NewSigner(private, pubX, pubY *big.Int, chainId ...string) (signer *Signer, err error) {
```

Comment:
This is an opinion so feel free to ignore. I generally prefer not to use named returns. It is harder to parse the logic, since they can be set arbitrarily and returned implicitly, which makes them more error prone

Reply:
I don't feel strongly either way, but consistency in the lib is important to me. 

Looks like I already had a mishmash of return syntax so, lets go forward with unnamed returns and weed named returns out when we come across them?